src/toil/jobStores/aws/jobStore.py

```diff
@@ -307,7 +307,8 @@ def _extractKeyInfoFromUrl(url):
         """
         s3 = boto.connect_s3()
         bucket = s3.get_bucket(url.netloc)
-        key = bucket.new_key(url.path[1:])
+        key = bucket.get_key(url.path[1:])
+        key = key if key is not None else bucket.new_key(url.path[1:])
```

Comment:
I don't like `createKey` because you're not really creating the key when you call `new_key`. You're just creating an instance of the Key class, not the remote key it represents. Bool args should always be keyword args and passed as such. How about

`_extractKeyInfoFromUrl(url, existing=None)`

If `existing` is `True` the key must exist, if existing is `False` it must not exist, otherwise it doesn't matter. 

The returned Key instance is never None but it may be unpopulated if the key is absent and `existing` is not True.